object/state.go

```diff
@@ -31,7 +36,63 @@ func (e *Environment) Len() int {
 	return len(e.store)
 }
 
+var baseInfo Map
+
+func (e *Environment) BaseInfo() Map {
+	if baseInfo != nil {
+		return baseInfo
+	}
+	baseInfo := make(Map, 6) // 5 here + all_ids
+	tokInfo := token.Info()
+	keys := make([]Object, 0, len(tokInfo.Keywords))
+	for _, v := range sets.Sort(tokInfo.Keywords) {
+		keys = append(keys, String{Value: v})
+	}
+	baseInfo[String{"keywords"}] = Array{Elements: keys}
+	keys = make([]Object, 0, len(tokInfo.Tokens))
+	for _, v := range sets.Sort(tokInfo.Tokens) {
+		keys = append(keys, String{Value: v})
+	}
+	baseInfo[String{"tokens"}] = Array{Elements: keys}
+	// Ditto cache this as it's set for a given environment.
+	ext := ExtraFunctions()
+	keys = make([]Object, 0, len(ext))
+	for k, _ := range ext {
+		keys = append(keys, String{Value: k})
+	}
+	arr := Array{Elements: keys}
+	sort.Sort(arr)
+	baseInfo[String{"gofuncs"}] = arr
+	baseInfo[String{"version"}] = String{Value: cli.ShortVersion}
+	baseInfo[String{"platform"}] = String{Value: cli.LongVersion}
+	return baseInfo
+}
+
+func (e *Environment) Info() Object {
+	allKeys := make(Map, e.depth)
+	for {
+		keys := make([]Object, 0, e.Len())
+		for k := range e.store {
+			keys = append(keys, String{Value: k})
+		}
+		arr := Array{Elements: keys}
+		sort.Sort(arr)
+		allKeys[Integer{Value: int64(e.depth)}] = arr
+		if e.outer == nil {
+			break
+		}
+		e = e.outer
+	}
+	info := e.BaseInfo()
+	info[String{"all_ids"}] = allKeys
+	// TODO: Cache this as it can't change without recompiling:
+	return info
+}
+
 func (e *Environment) Get(name string) (Object, bool) {
+	if name == "info" {
```

Comment:
Do you plan to be able to use info with argument, such as "info sin"

Or "help sin"

Reply:
functions being objects, you don't need "help sin" you can just type "sin":

<img width="114" alt="Screenshot 2024-08-05 at 12 45 58 PM" src="https://github.com/user-attachments/assets/42e57d3e-fd62-4b95-a8d5-f2e966212db4">

and to repeat it's a map; eg info["gofuncs"] is just the go function extensions array

Reply:
we can add help text later though, that could be useful (keyed of that map)

Reply:
#108